### ISSUE TITLE
 Include conditions in proxying functionality

### ISSUE THREAD

**Issue by Alan-Vlc** [wiremock/wiremock#2695]
### Proposal

Hello, 
I am testing Wiremock's proxying functionality to “route” certain http requests that arrive to my Wiremock pod to a URL. Everything works correctly using the following configuration file: {
  
```
  {
  "mappings": [
    {
      "request": {
        "urlPathPattern": "/stores/products/out-of-stock-store-lost-sales/.*"
      },
      "response": {
        "proxyBaseUrl": "https://myendpoint.testing”
        
      }
    }
  ]
}
```

Would it be possible to establish conditions for the proxying functionality? Some examples:

* 50% of requests that meet "urlPathPattern" https are redirected to “proxyBaseUrl” and the remaining 50% of http requests return a 500 response code.
* 30% of requests that match "urlPathPattern" https are redirected to “proxyBaseUrl”, another 30% of http requests return a 500 response code and the remaining 40% of http requests return a 401 response code.

I would appreciate it if you considered implementing this type of improvement in future versions.

Thanks in advance

**Comment by tomakehurst** [comment]
I don't think this is something we'd want to do in the core library, but you could implement this fairly trivially by building a `ResponseTransformerV2` extension.

**Comment by Alan-Vlc** [comment]
Thanks for your response.

Sorry, but I don't quite understand how to cover the need with what you indicate. I have followed the following link ( https://wiremock.org/docs/extensibility/transforming-responses/) and I don't understand how to cover the examples I have specified.
Keep in mind that I build a pod with wiremock to act as a proxy, since we do not want to put java code in the application part.

Thanks

**Comment by tomakehurst** [comment]
I suggest you head over to the Slack community if you need some support creating and installing WireMock extensions.

**Comment by tomakehurst** [comment]
No, webhooks is built into the core now, so there's no extension in more recent versions
